### ISSUE TITLE
[REQUEST] RegexMatch FUNCTION capturing group

### ISSUE THREAD

**Issue by dNitro** [openbullet/openbullet#426]
**Is your feature request related to a problem? Please describe.**
Need a way of capturing groups in RegexMatch FUNCTION like what is implemented in PARSE > REGEX

**Describe the solution you'd like**
An **output** textbox so we could capture regex groups: [0]: whole match, [1]: first group and ... just like PARSE > REGEX

**Describe alternatives you've considered**
Someone could use PARSE > REGEX to achieve intended results but in some situations using FUNCTION MatchRegex is much more clear

**Additional context**
As you can see **Output** input box is absent in FUNCTION RegexMatch:
![Screen Shot 2019-12-02 at 04 42 53](https://user-images.githubusercontent.com/13446672/69948961-8ec77700-14be-11ea-9562-d8f91fa02c59.png)
![Screen Shot 2019-12-02 at 04 43 05](https://user-images.githubusercontent.com/13446672/69948967-9129d100-14be-11ea-85d3-0740e584281a.png)

**Comment by openbullet** [comment]
Yes I was already thinking about adding this, thanks!

**Comment by openbullet** [comment]
Sorry I was thinking about this again, what would be the advantage of having the same exact functionality as the PARSE block replicated inside the FUNCTION block?

**Comment by dNitro** [comment]
@openbullet, Of course, there is no advantage! To me, this RegexMatch FUNCTION seems redundant as PARSE > Regex is more versatile. I was thinking that only FUNCTIONS could loop over Lists ( i.e. when input is: `<ITEMS[*]>` for example ) but I also set the input as `<ITEMS[*]>` in a PARSE block and it extracts each subgroup from each item using Regex option.

OB could get rid of this FUNCTION, but it's a breaking change; some configs may already using it; Then a better option would be to make it also match subgroups with default match to all.

Anyway, subgroup matching is a reasonable expectation of a Regex function. Already this function is imperfect and of course, having an alternative option ( i.e. PARSE block ) makes it *low priority* to implement. ^-^

**Comment by openbullet** [comment]
Yes I cannot get rid of existing functions because it would break existing configs. I will label this as low prio and we'll see.
